pkg/cmd/checkendpoints/controller/pod_network_connectivity_check_controller.go

```diff
@@ -0,0 +1,108 @@
+type controller struct {
+	factory.Controller
+	podName      string
+	podNamespace string
+	checksGetter operatorcontrolplaneclientv1alpha1.PodNetworkConnectivityCheckInterface
+	checkLister  v1alpha1.PodNetworkConnectivityCheckNamespaceLister
+	recorder     events.Recorder
+	// each PodNetworkConnectivityCheck gets its own ConnectionChecker
+	updaters map[string]ConnectionChecker
+}
+
+// Returns a new PodNetworkConnectivityCheckController that performs network connectivity checks
+// as specified in the PodNetworkConnectivityChecks defined in the specified namespace, for the specified pod.
+func NewPodNetworkConnectivityCheckController(podName, podNamespace string, checksGetter operatorcontrolplaneclientv1alpha1.PodNetworkConnectivityChecksGetter, checkInformer alpha1.PodNetworkConnectivityCheckInformer, recorder events.Recorder) PodNetworkConnectivityCheckController {
+	c := &controller{
+		podName:      podName,
+		podNamespace: podNamespace,
+		checksGetter: checksGetter.PodNetworkConnectivityChecks(podNamespace),
+		checkLister:  checkInformer.Lister().PodNetworkConnectivityChecks(podNamespace),
+		recorder:     recorder,
+		updaters:     map[string]ConnectionChecker{},
+	}
+	c.Controller = factory.New().
+		WithSync(c.Sync).
+		WithBareInformers(checkInformer.Informer()).
+		ResyncEvery(1*time.Second).
+		ToController("check-endpoints", recorder)
+	return c
+}
+
+// Sync ensures that the status updaters for each PodNetworkConnectivityCheck is started
+// and then performs each check.
+func (c *controller) Sync(ctx context.Context, syncContext factory.SyncContext) error {
+	checkList, err := c.checkLister.List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	// filter list of checks for current pod
+	var checks []*operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck
+	for _, check := range checkList {
+		if check.Spec.SourcePod == c.podName {
+			checks = append(checks, check)
+		}
+	}
+
+	// create & start status updaters if needed
+	for _, check := range checks {
+		if updater := c.updaters[check.Name]; updater == nil {
+			c.updaters[check.Name] = NewConnectionChecker(check, c, c.recorder)
+			c.updaters[check.Name].Run(ctx)
+		}
+	}
+
+	// stop & delete unneeded status updaters
+	for name, updater := range c.updaters {
+		var keep bool
+		for _, check := range checks {
+			if check.Name == name {
+				keep = true
+				break
+			}
+		}
+		if !keep {
+			updater.Stop()
+			delete(c.updaters, name)
+		}
+	}
+
+	return nil
+}
+
+// Get implements PodNetworkConnectivityCheckClient
+func (c *controller) Get(name string) (*operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck, error) {
+	return c.checkLister.Get(name)
+}
+
+// UpdateStatus implements v1alpha1helpers.PodNetworkConnectivityCheckClient
+func (c *controller) UpdateStatus(ctx context.Context, check *operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck, opts metav1.UpdateOptions) (*operatorcontrolplanev1alpha1.PodNetworkConnectivityCheck, error) {
```

Comment:
is this used?

Reply:
Yes. Implements **PodNetworkConnectivityCheckClient** client that **ConnectionChecker** uses.